File: data-structures/arrays/MyArray.java
// array declaration and operations

public class MyArray {
    private int [] a;
    private int n;
    
    public MyArray() {
        a = new int[12];
        n = 0;
    }
    
    public void initialize() {
        for (int i = 0; i < 10; i++) {
            a[i] = i+1;
        }
        n = 10;
    }
    
    // print the array
    public void print() {
        for (int i = 0; i < n; i++) {
            System.out.print(a[i] + " ");
        }
        System.out.println();
    }
    
    public int get(int i) {
        return a[i];
    }
    
    public void set(int position, int value) {
        a[position] = value;
    }
    
    public void insert(int position, int value) {
        for (int i = n; i >= position; i--) {
            a[i + 1] = a[i];
        }
        a[position] = value;
        n = n + 1;
    }
    
    public void delete(int position) {
        for(int i = 0; i < n; i++) {
            a[i] = a[i + 1];
        }
        n = n - 1;
    }
    
    public static void main(String [] args) {
        MyArray arr = new MyArray();
        
        // initialize the array (takes O(n) time)
        arr.initialize();
        
        // print to verify 
        arr.print(); // prints 1 2 3 4 5 6 7 8 9 10 
        
        // read the item in 5th position (position starts from 0)
        System.out.println(arr.get(5)); // prints 6
        
        // write 12 in first position
        arr.set(0, 12); // prints 12 2 3 4 5 6 7 8 9 10 
        
        // print to verify
        arr.print(); // 12 2 3 4 5 6 7 8 9 10 
        
        // insert 100 in first position
        arr.insert(0, 100); // 100 12 2 3 4 5 6 7 8 9 10
        
        // print to verify 
        arr.print(); 
        
        // remove item in first position
        arr.delete(0); // 12 2 3 4 5 6 7 8 9 10
        
        // print to verify
        arr.print();
    }
}

File: data-structures/arrays/MyArray.cpp
// array declaration and operations
#include <iostream>

using namespace std;

// prints the array
void print(int a[], int n) {
    int i;
    for (i = 0; i < n; i++) {
        cout<<a[i]<<" ";
    }
    cout<<endl;
}

int main() {
    int i;
    // declares initializes 1d array of 12 integers
    int a[12];
    
    // keeps track of current number of items in the array
    int n = 0;
    
    // initialize first 10 items (takes O(n) time) 
    for (i = 0; i < 10; i++) {
        a[i] = i + 1;
        n = n + 1;
    }
    
    // print and see if it has all 10 items
    print(a, n); // prints 1, 2, 3, 4, 5, 6, 7, 8, 9, 10
    
    // read the item in 5th position (position starts from 0)
    cout<<a[5]<<endl; // prints 6
    
    // write 12 in first position
    a[0] = 12; 
    
    // print to see the change
    print(a, n); // prints 12, 2, 3, 4, 5, 6, 7, 8, 9, 10
    
    // insert 100 to first position
    for (i = n; i >= 0; i--) {
        a[i + 1] = a[i]; // move one step to right
    }
    
    a[0] = 100; // write 100 in first position
    n = n + 1;
    
    // print to verify
    print(a, n); // prints 100 12 2 3 4 5 6 7 8 9 10 
    
    // remove the item in the first position
    for(i = 0; i < n; i++) {
        a[i] = a[i + 1];
    }
    
    n = n - 1;
    
    // print to verify
    print(a, n); // prints 12 2 3 4 5 6 7 8 9 10 
    
	return 0;
}